wallet-new/src/Cardano/Wallet/API/V1/Accounts.hs

```diff
@@ -31,4 +34,16 @@ type API
           :> Summary "Update an Account for the given Wallet."
           :> ReqBody '[ValidJSON] (Update Account)
           :> Put '[ValidJSON] (WalletResponse Account)
+    :<|> "wallets" :> CaptureWalletId :> "accounts"
+          :> CaptureAccountId :> "addresses"
+          :> Summary "Retrieve only account's addressees."
+          :> WalletRequestParams
+          :> FilterBy '[V1 Core.Address] WalletAddress
+          :> SortBy '[V1 Core.Address] WalletAddress
```

Comment:
Does sorting on address really make sense? They are just a bunch of base58-encoded strings after all, so I'm wondering whether or not we should allow sorting, as it buys virtually nothing 😉

Reply:
It allows pagination based on an indexed part of the datatype, which makes for super efficient querying.

Reply:
> It allows pagination based on an indexed part of the datatype, which makes for super efficient querying.

@parsonsmatt @KtorZ  Isn't what `FilterBy` buys you already? Unless I am mistaken, note that the type-level sorting & filtering has no effect on the (already-defined) IxSet indices 😉 Once those are defined (as in `instance Indexable ...`) which one you use for filter & sort (or the lack thereof) really doesn't undermine querying efficiency 😉 

Imho keeping `SortBy` here is a mistake, as it would encourage exchanges to sort by addresses, which is an operation terribly wasteful, as there is no meaningful ordering for addresses, as the lexicographical order doesn't really help here. My 2 cents anyway, very happy to be proved wrong! 😉

Reply:
In a way, pagination without ordering doesn't make much sense to me, does it? Do we even have guarantees with the acid-state db that we will yield the same elements from two successive get queries? Or is it completely non-deterministic on non-indexed items?

Though, I'd rather follow your intuition on that @adinapoli-iohk. Note that the PR has been merged in a feature-branch and not yet to develop, so there's still time to do some minor adjusments!

Reply:
Haven't followed the details here but I agree that for pagination it is important to have an indexed attribute available to sort on. Not sure how much control we need to give the user over this, since as @adinapoli-iohk points out, sorting on base58 strings isn't particularly meaningful. Moreover, in the new wallet we clearly distinguish between the primary key and the rest of the data structure and don't require Ord instances for the whole type; in this case, the natural way to sort by "address" would be to sort on its key (which is basically just the pair of integers telling it how that address is derived from the root). So I think it's important that "sort by address" doesn't necessarily mean "sort lexicographically", but rather "give me some kind of pagination I can depend on" -- so that we are free to switch over the sorting to the primary key once we move to the new wallet, rather than being forced to still support this lexicographical sort.

Reply:
> In a way, pagination without ordering doesn't make much sense to me, does it?

Indeed, but _I think_ we should be fine in this case, as in the current implementation we simply call `IxSet.toList` which, AFAIK, uses descending ordering, which is a reasonable default, I believe.

My main gripe was that, if we leave that `SortBy` as-it-is, we are really exposing `sort_by=address[DES|ASC]` to the API, which means exchanges could enforce an explicit sorting on the addresses, which, considering the amount of addresses an exchanges has, I would prefer to not pay the explicit price, if possible 😀 

Am I being mental? 😛

Reply:
I usually say: If users can do something, eventually they will. I wonder if we shouldn't indeed remove that sorting for now, until we have something more meaningful we can actually sort on.

Reply:
> If users can do something, eventually they will.

Aye, that is usually my experience -- that's why I have flagged this up 😉

Reply:
NOTE: Sorting was removed on the main story branch